server/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -536,6 +546,46 @@ public static GeoPoint parseGeoPoint(XContentParser parser, GeoPoint point, fina
         }
     }
 
+    private static double parseValidDouble(XContentSubParser subParser, String field) throws IOException {
+        try {
+            return switch (subParser.currentToken()) {
+                case VALUE_NUMBER, VALUE_STRING -> subParser.doubleValue(true);
+                default -> throw new ElasticsearchParseException("{} must be a number", field);
```

Comment:
Maybe we should keep the square brackets?
```suggestion
                default -> throw new ElasticsearchParseException("[{}] must be a number", field);
```

Reply:
fair enough